sdk/core/core-amqp/src/connectionConfig/eventhubConnectionConfig.ts

```diff
@@ -104,15 +104,15 @@ export module EventHubConnectionConfig {
     };
 
     (config as EventHubConnectionConfig).getSenderAudience = (partitionId?: string | number) => {
-      if (partitionId != undefined) {
+      if (partitionId !== undefined) {
```

Comment:
We have intentionally used != in many places to cover both `undefined` and `null`
So, this change would result in change in behavior when `partitionid` is null

Reply:
Lets remove the related rule and have the library owners deal with this separately

Reply:
Thank you. We should use strict comparisons instead! Please let me know what should also be changed to check for `undefined` and `null`.

Reply:
I'm up for doing these changes myself as part of this PR. I'll have a follow up to this feedback later in the week. Though it can be challenging to spot all places, so any hint is appreciated 🌞

Reply:
I would really recommend not doing these changes in this PR and keeping this PR restricted to the most simplest of changes that we can do. The package owners can do a follow up based on the issue you logged

Reply:
yeah I think for this particular rule  we should have a tracking bug. 

It also seems reasonable that one would want to check for the empty string here but not the number 0, for example (unless partition ids are 1-indexed.)

Reply:
I'll revert the `=== undefined` and `!== undefined` changes